src/main/java/cubicchunks/worldgen/generator/flat/FlatTerrainProcessor.java

```diff
@@ -23,43 +23,74 @@
  */
 package cubicchunks.worldgen.generator.flat;
 
-import cubicchunks.util.processor.CubeProcessor;
+import cubicchunks.world.ICubicWorld;
 import cubicchunks.world.cube.Cube;
+import cubicchunks.worldgen.generator.BasicCubeGenerator;
+import cubicchunks.worldgen.generator.CubePrimer;
+import cubicchunks.worldgen.generator.ICubePrimer;
+import net.minecraft.entity.boss.EntityWither;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3i;
+import net.minecraft.world.World;
 
-public class FlatTerrainProcessor implements CubeProcessor {
+public class FlatTerrainProcessor extends BasicCubeGenerator {
 
-	@Override public void calculate(Cube cube) {
-		if (cube.getY() >= 0) {
-			return;
+	public FlatTerrainProcessor(ICubicWorld world){
+		super(world);
+	}
+
+	@Override
+	public ICubePrimer generateCube(int cubeX, int cubeY, int cubeZ) {
+		ICubePrimer primer = new CubePrimer();
+
+		if (cubeY >= 0) {
+			return primer;
 		}
-		if (cube.getY() == -1) {
-			BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos();
+		if (cubeY == -1) {
 			for (int x = 0; x < 16; x++) {
 				for (int z = 0; z < 16; z++) {
-					pos.setPos(x, 15, z);
-					cube.setBlockForGeneration(pos, Blocks.GRASS.getDefaultState());
+					primer.setBlockState(x, 15, z, Blocks.GRASS.getDefaultState());
 					for (int y = 14; y >= 10; y--) {
-						pos.setPos(x, y, z);
-						cube.setBlockForGeneration(pos, Blocks.DIRT.getDefaultState());
+						primer.setBlockState(x, y, z, Blocks.DIRT.getDefaultState());
 					}
 					for (int y = 9; y >= 0; y--) {
-						pos.setPos(x, y, z);
-						cube.setBlockForGeneration(pos, Blocks.STONE.getDefaultState());
+						primer.setBlockState(x, y, z, Blocks.STONE.getDefaultState());
 					}
 				}
 			}
-			return;
+			return primer;
 		}
-		BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos();
+
 		for (int x = 0; x < 16; x++) {
 			for (int z = 0; z < 16; z++) {
 				for (int y = 0; y < 16; y++) {
-					pos.setPos(x, y, z);
-					cube.setBlockForGeneration(pos, Blocks.STONE.getDefaultState());
+					primer.setBlockState(x, y, z, Blocks.STONE.getDefaultState());
 				}
 			}
 		}
+
+		return primer;
+	}
+
+	@Override
+	public void populate(Cube cube) {
+		if(cube.containsBlockPos(new BlockPos(700, 100, 200))){
+			for(int i = 0;i < 60;i++){
+				EntityWither wither = new EntityWither((World)world);
+				wither.setPositionAndRotation(700.0, 100.0, 200.0, 0.0f, 0.0f);
+				world.spawnEntityInWorld(wither);
```

Comment:
Remember when I was testing to see if withers where able to shoot skulls? This was part of that